eo-maven-plugin/src/main/java/org/eolang/maven/OyRemote.java

```diff
@@ -42,25 +45,16 @@ public final class OyRemote implements Objectionary {
     private final String template;
 
     /**
-     * Ctor.
+     * Constructor.
+     * @param tag Tag
+     * @throws IOException if fails
+     * @throws IllegalArgumentException if fails
      */
-    public OyRemote() {
-        this("master");
-    }
-
-    /**
-     * Ctor.
-     * @param hash The GitHub hash
-     * @todo #490:30m Resolve abbreviated hash to a proper hash.
-     *  In order to avoid collisions resolve hash
-     *  (or branch) to a complete sha-256 hash of the commit.
-     *  Use only sha-256 hashes as a caching criteria.
-     */
-    public OyRemote(final String hash) {
+    public OyRemote(final String tag) throws IOException, IllegalArgumentException {
         this.template = String.format(
             // @checkstyle LineLength (1 line)
             "https://raw.githubusercontent.com/objectionary/home/%s/objects/%%s.eo",
-            hash
+            new HashOfTag(tag).getHash()
```

Comment:
@yasamprom if you rename `getHash` to `toString`, you would not need to call it here. It would be called behind the scene

Reply:
@yegor256 `toString()` should't throw exceptions. We can change structure and avoid rethrowing exceptions. Should we do it?